lib/src/state/dna_design.dart

```diff
@@ -553,7 +553,14 @@ abstract class DNADesign with UnusedFields implements Built<DNADesign, DNADesign
 
   bool has_nondefault_max_offset(Helix helix) {
     var ends = domains_on_helix(helix.idx).map((ss) => ss.end);
-    int max_end = ends.isEmpty ? constants.default_max_offset : ends.reduce(max);
+    int max_end = 0;
```

Comment:
I don't think this works. Previously, the default `max_offset` for a helix depended only on that helix (`constants.default_max_offset` if the helix had no domains, and the maximum end offset of any domain on it otherwise).

Now, the default max_offset is a function of the whole design. This would fail when writing a design like the first one in the issue (#351):

![image](https://user-images.githubusercontent.com/19274365/86795896-e153ad80-c022-11ea-8ba2-c9d225693156.png)

It would assume all of the helices have their default value, since they all end just after the last domain, but in fact those helices (other than -9 and -10) do *not* have the default value, so they should have an explicit `max_offset` field written in the JSON. The helices below have the default value:

![image](https://user-images.githubusercontent.com/19274365/86797437-8327ca00-c024-11ea-8819-85f21231fbe5.png)

So although the method takes a helix index as input, it cannot reference *only* that helix. (It will be a bit wasteful since those values will be recalculated on every helix, but that's fine.)